ckanext/datastore/writer.py

```diff
@@ -3,13 +3,39 @@
 from contextlib import contextmanager
 from email.utils import encode_rfc2231
 from simplejson import dumps
+import plugins
 from xml.etree.cElementTree import Element, SubElement, ElementTree
+from ckanext.datastore.interfaces import IWriter
 
 import unicodecsv
 
 from codecs import BOM_UTF8
 
 
+def get_writers(fmt):
+
+    for plugin in plugins.PluginImplementations(IWriter):
+        if hasattr(plugin, u'get_writer'):
+            plugin_writer = plugin.get_writer()
+
+    if fmt == u'csv':
+        writer_factory = csv_writer
+        records_format = u'csv'
+    elif fmt == u'tsv':
+        writer_factory = tsv_writer
+        records_format = u'tsv'
+    elif fmt == u'json':
+        writer_factory = json_writer
+        records_format = u'lists'
+    elif fmt == u'xml':
+        writer_factory = xml_writer
+        records_format = u'objects'
+    else:
+        writer_factory = plugin_writer
+        records_format = u'objects'
```

Comment:
why wouldn't a writer plugin want to be able to use one of the more efficient dump formats?